lib/spack/llnl/util/filesystem.py

```diff
@@ -583,6 +584,54 @@ def __str__(self):
         return self.joined()
 
 
+def find_system_libraries(args, shared=True):
+    """Searches the usual system library locations for the libraries
+    specified in args.
+
+    Search order is as follows:
+
+    1. /lib64
+    2. /lib
+    3. /usr/lib64
+    4. /usr/lib
+    5. /usr/local/lib64
+    6. /usr/local/lib
+
+    :param args: Library name(s) to search for
+    :type args: str or collections.Sequence
+    :param bool shared: if True searches for shared libraries,
+
+    :returns: The libraries that have been found
+    :rtype: LibraryList
+    """
+    if isinstance(args, str):
+        args = [args]
+    elif not isinstance(args, collections.Sequence):
+        message = '{0} expects a string or sequence of strings as the '
+        message += 'first argument [got {1} instead]'
+        message = message.format(find_system_libraries.__name__, type(args))
+        raise TypeError(message)
+
+    libraries_found = []
+    search_locations = [
+        '/lib64',
+        '/lib',
+        '/usr/lib64',
+        '/usr/lib',
+        '/usr/local/lib64',
```

Comment:
probably the last two rarely contain things like `libpthread` or `libm`. But since they are last in the list, i think it's fine to keep it as-is.